src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableSchemaObject.java

```diff
@@ -1,22 +1,98 @@
 
   public static final SchemaObjectType TYPE = SchemaObjectType.TABLE;
 
-  public TableSchemaObject(TableMetadata tableMetadata) {
+  private final List<VectorConfig> vectorConfigs;
+
+  private TableSchemaObject(TableMetadata tableMetadata, List<VectorConfig> vectorConfigs) {
     super(TYPE, tableMetadata);
+    this.vectorConfigs = vectorConfigs;
   }
 
   @Override
-  public VectorConfig vectorConfig() {
-    return VectorConfig.notEnabledVectorConfig();
+  public List<VectorConfig> vectorConfigs() {
+    return vectorConfigs;
   }
 
   @Override
   public IndexUsage newIndexUsage() {
     return IndexUsage.NO_OP;
   }
+
+  /**
+   * Get table schema object from table metadata
+   *
+   * @param tableMetadata
+   * @param objectMapper
+   * @return
+   */
+  public static TableSchemaObject getTableSettings(
+      TableMetadata tableMetadata, ObjectMapper objectMapper) {
+    Map<String, String> extensions =
+        (Map<String, String>)
+            tableMetadata.getOptions().get(CqlIdentifier.fromInternal("extensions"));
+    String vectorize = extensions != null ? extensions.get("vectorize") : null;
+    Map<String, VectorConfig.VectorizeConfig> resultMap = new HashMap<>();
+    if (vectorize != null) {
+      String vectorizeJson = new String(ByteUtils.fromHexString(vectorize).array());
+      // Define the TypeReference for Map<String, VectorConfig.VectorizeConfig>
+      TypeReference<Map<String, VectorConfig.VectorizeConfig>> typeRef =
+          new TypeReference<Map<String, VectorConfig.VectorizeConfig>>() {};
+
+      // Convert JSON string to Map
+      try {
+        resultMap = objectMapper.readValue(vectorizeJson, typeRef);
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    List<VectorConfig> vectorConfigs = new ArrayList<>();
+    for (Map.Entry<CqlIdentifier, ColumnMetadata> column : tableMetadata.getColumns().entrySet()) {
+      if (column.getValue().getType() instanceof VectorType vectorType) {
+        final Optional<IndexMetadata> index = tableMetadata.getIndex(column.getKey());
+        SimilarityFunction similarityFunction = SimilarityFunction.COSINE;
+        if (index.isPresent()) {
+          final IndexMetadata indexMetadata = index.get();
+          final Map<String, String> indexOptions = indexMetadata.getOptions();
+          final String similarityFunctionValue = indexOptions.get("similarity_function");
+          if (similarityFunctionValue != null) {
+            similarityFunction = SimilarityFunction.fromString(similarityFunctionValue);
+          }
+        }
+        int dimension = vectorType.getDimensions();
+        VectorConfig vectorConfig =
+            new VectorConfig(
+                true,
+                column.getKey().asInternal(),
+                dimension,
+                similarityFunction,
+                resultMap.get(column.getKey().asInternal()));
+        vectorConfigs.add(vectorConfig);
```

Comment:
So we are expecting error configs to be ordered in Map/JSON Object? Do we specify ordering when building the Map/Object?

Reply:
We need to maintain ordering for the columns. The configuration has the field (column) name in it. FOr collection it will be set as `$vectorize`

Reply:
Ok so we'll match by name tho, as necessary, not necessarily by index?

Reply:
Yes.